ocs_ci/ocs/quay_operator.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from time import sleep
+
+from ocs_ci.helpers import helpers
+from ocs_ci.helpers.helpers import storagecluster_independent_check
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import TimeoutSampler, exec_cmd, run_cmd
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+
+logger = logging.getLogger(__name__)
+
+
+class QuayOperator(object):
+    """
+    Quay operator class
+
+    """
+
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.namespace = "openshift-operators"
+        self.quay_operator = None
+        self.quay_registry = None
+        self.quay_pod_obj = OCP(kind="pod", namespace=self.namespace)
+        self.quay_registry_name = ""
+        self.quay_operator_csv = ""
+        self.sc_default = False
+        self.sc_name = (
+            constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_RBD
+            if storagecluster_independent_check()
+            else constants.DEFAULT_STORAGECLASS_RBD
+        )
+
+    def setup_quay_operator(self):
+        """
+        Deploys Quay operator
+
+        """
+        quay_operator_data = templating.load_yaml(file=constants.QUAY_SUB)
+        self.quay_operator_csv = quay_operator_data["spec"]["startingCSV"]
+        self.quay_operator = OCS(**quay_operator_data)
+        logger.info(f"Installing Quay operator: {self.quay_operator.name}")
+        self.quay_operator.create()
+        for quay_pod in TimeoutSampler(
+            300, 10, get_pod_name_by_pattern, "quay-operator", self.namespace
+        ):
+            if quay_pod:
+                assert self.quay_pod_obj.wait_for_resource(
+                    condition=constants.STATUS_RUNNING,
+                    resource_name=quay_pod[0],
+                    sleep=30,
+                    timeout=600,
+                )
+                break
+
+    def create_quay_registry(self):
+        """
+        Creates Quay registry
+
+        """
+        if not helpers.get_default_storage_class():
+            patch = ' \'{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}\' '
+            run_cmd(
+                f"oc patch storageclass {self.sc_name} "
+                f"-p {patch} "
+                f"--request-timeout=120s"
+            )
+            self.sc_default = True
```

Comment:
we already have an existing function to change the default storage class to a given SC.
change_default_storageclass() in helpers.py file, please use it here.

Reply:
I am not trying to change the default storageclass. I am just checking if there is an existing default storageclass, if so use or patch a storageclass.

Reply:
ok, can we use default_storage_class here?

Reply:
Don't think so. Default_storage_class gets sc object based on the interface type given which is not required here.